'make html' with Python 3.5.4 causes an EOFError

Subject: This seems to be related to multiprocess

### Problem
- Parsing a file created by sphinx-apidoc in multiprocess
- EOF error is raised.
- The file works fine on python 3.5.2, with no other differences.

#### Procedure to reproduce the problem
Download my attached example project and call:
```
make docker_docs
```
This will pull the python 3.5.4 docker image and expose the bug. It does leave hte docker image running.


#### Error logs / results

Edit: replace logs with the logs actually gotten by running 'make docker_docs'
```
rm -rf /tmp/mountable_example
cp -r /tmp/reproduce /tmp/mountable_example
3.5: Pulling from library/python
Digest: sha256:dab79666dd592b82aafd1e0d97c0ac0bbddcb77b5780680a28bc2707b72d244b
Status: Image is up to date for python:3.5
python setup.py develop
running develop
running egg_info
writing top-level names to example.egg-info/top_level.txt
writing example.egg-info/PKG-INFO
writing dependency_links to example.egg-info/dependency_links.txt
reading manifest file 'example.egg-info/SOURCES.txt'
writing manifest file 'example.egg-info/SOURCES.txt'
running build_ext
Creating /usr/local/lib/python3.5/site-packages/example.egg-link (link to .)
Adding example 0.1.0 to easy-install.pth file

Installed /reproduce
Processing dependencies for example==0.1.0
Finished processing dependencies for example==0.1.0
pip install Sphinx==1.6.3 sphinx_rtd_theme
Collecting Sphinx==1.6.3
  Downloading Sphinx-1.6.3-py2.py3-none-any.whl (1.9MB)
Collecting sphinx_rtd_theme
  Downloading sphinx_rtd_theme-0.2.4-py2.py3-none-any.whl (1.4MB)
Collecting sphinxcontrib-websupport (from Sphinx==1.6.3)
  Downloading sphinxcontrib_websupport-1.0.1-py2.py3-none-any.whl
Collecting requests>=2.0.0 (from Sphinx==1.6.3)
  Downloading requests-2.18.4-py2.py3-none-any.whl (88kB)
Collecting six>=1.5 (from Sphinx==1.6.3)
  Downloading six-1.10.0-py2.py3-none-any.whl
Collecting snowballstemmer>=1.1 (from Sphinx==1.6.3)
  Downloading snowballstemmer-1.2.1-py2.py3-none-any.whl (64kB)
Requirement already satisfied: setuptools in /usr/local/lib/python3.5/site-packages (from Sphinx==1.6.3)
Collecting Pygments>=2.0 (from Sphinx==1.6.3)
  Downloading Pygments-2.2.0-py2.py3-none-any.whl (841kB)
Collecting Jinja2>=2.3 (from Sphinx==1.6.3)
  Downloading Jinja2-2.9.6-py2.py3-none-any.whl (340kB)
Collecting babel!=2.0,>=1.3 (from Sphinx==1.6.3)
  Downloading Babel-2.4.0-py2.py3-none-any.whl (6.8MB)
Collecting alabaster<0.8,>=0.7 (from Sphinx==1.6.3)
  Downloading alabaster-0.7.10-py2.py3-none-any.whl
Collecting docutils>=0.11 (from Sphinx==1.6.3)
  Downloading docutils-0.14-py3-none-any.whl (543kB)
Collecting imagesize (from Sphinx==1.6.3)
  Downloading imagesize-0.7.1-py2.py3-none-any.whl
Collecting certifi>=2017.4.17 (from requests>=2.0.0->Sphinx==1.6.3)
  Downloading certifi-2017.7.27.1-py2.py3-none-any.whl (349kB)
Collecting urllib3<1.23,>=1.21.1 (from requests>=2.0.0->Sphinx==1.6.3)
  Downloading urllib3-1.22-py2.py3-none-any.whl (132kB)
Collecting chardet<3.1.0,>=3.0.2 (from requests>=2.0.0->Sphinx==1.6.3)
  Downloading chardet-3.0.4-py2.py3-none-any.whl (133kB)
Collecting idna<2.7,>=2.5 (from requests>=2.0.0->Sphinx==1.6.3)
  Downloading idna-2.6-py2.py3-none-any.whl (56kB)
Collecting MarkupSafe>=0.23 (from Jinja2>=2.3->Sphinx==1.6.3)
  Downloading MarkupSafe-1.0.tar.gz
Collecting pytz>=0a (from babel!=2.0,>=1.3->Sphinx==1.6.3)
  Downloading pytz-2017.2-py2.py3-none-any.whl (484kB)
Building wheels for collected packages: MarkupSafe
  Running setup.py bdist_wheel for MarkupSafe: started
  Running setup.py bdist_wheel for MarkupSafe: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/88/a7/30/e39a54a87bcbe25308fa3ca64e8ddc75d9b3e5afa21ee32d57
Successfully built MarkupSafe
Installing collected packages: sphinxcontrib-websupport, certifi, urllib3, chardet, idna, requests, six, snowballstemmer, Pygments, MarkupSafe, Jinja2, pytz, babel, alabaster, docutils, imagesize, Sphinx, sphinx-rtd-theme
Successfully installed Jinja2-2.9.6 MarkupSafe-1.0 Pygments-2.2.0 Sphinx-1.6.3 alabaster-0.7.10 babel-2.4.0 certifi-2017.7.27.1 chardet-3.0.4 docutils-0.14 idna-2.6 imagesize-0.7.1 pytz-2017.2 requests-2.18.4 six-1.10.0 snowballstemmer-1.2.1 sphinx-rtd-theme-0.2.4 sphinxcontrib-websupport-1.0.1 urllib3-1.22
rm -rf build
make --directory docs/ clean
make[1]: Entering directory '/reproduce/docs'
rm -rf _build/* reference/*
make[1]: Leaving directory '/reproduce/docs'
rm -fr docs/reference
mkdir -p docs/reference docs/_static
for m in project; do \
   sphinx-apidoc -e -o docs/reference $m ; \
done
Creating file docs/reference/project.example.rst.
Creating file docs/reference/project.rst.
Creating file docs/reference/modules.rst.
rm -rf docs/reference/modules.rst
make --directory docs/ html
make[1]: Entering directory '/reproduce/docs'
sphinx-build -b html -d _build/doctrees  -W -j 4 -n   . _build/html
Running Sphinx v1.6.3
making output directory...
loading pickled environment... not yet created
loading intersphinx inventory from http://docs.python.org/3/objects.inv...
intersphinx inventory has moved: http://docs.python.org/3/objects.inv -> https://docs.python.org/3/objects.inv
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 3 source files that are out of date
updating environment: 3 added, 0 changed, 0 removed
reading sources... [ 33%] index
reading sources... [ 66%] reference/project
reading sources... [100%] reference/project.example

looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [ 50%] reference/project
writing output... [100%] reference/project.example

waiting for workers...

Exception occurred:
  File "/usr/local/lib/python3.5/multiprocessing/connection.py", line 383, in _recv
    raise EOFError
EOFError
The full traceback has been saved in /tmp/sphinx-err-y75v1yhp.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make[1]: *** [html] Error 1
Makefile:54: recipe for target 'html' failed
make[1]: Leaving directory '/reproduce/docs'
make: *** [docs] Error 2
Makefile:7: recipe for target 'docs' failed
Makefile:27: recipe for target 'docker_docs' failed
make: *** [docker_docs] Error 2
```

#### Expected results
Don't fail on an EOFError, build should succeed on reproducible project.

### Reproducible project / your project
- [reproduce.zip](https://github.com/sphinx-doc/sphinx/files/1226304/reproduce.zip)

### Environment info
- OS: Linux
- Python version: 3.5.4
- Sphinx version: 1.6.3

--- a/CHANGES
+++ b/CHANGES
@@ -26,6 +26,7 @@ Bugs fixed
 * Missing ``texinputs_win/Makefile`` to be used in latexpdf builder on windows.
 * #4026: nature: Fix macOS Safari scrollbar color
 * #3877: Fix for C++ multiline signatures.
+* #4006: Fix crash on parallel build
 
 Testing
 --------

--- a/sphinx/util/parallel.py
+++ b/sphinx/util/parallel.py
@@ -122,6 +122,7 @@ class ParallelTasks(object):
                     logger.handle(log)
                 self._result_funcs.pop(tid)(self._args.pop(tid), result)
                 self._procs[tid].join()
+                self._precvs.pop(tid)
                 self._pworking -= 1
                 break
         else:

